x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/get_snapshot_helper.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { MonitorGroups, MonitorGroupIterator } from './search';
+
+export const getSnapshotCountHelper = async (iterator: MonitorGroupIterator) => {
+  const items: MonitorGroups[] = [];
+  let res: MonitorGroups | null;
+  // query the index to find the most recent check group for each monitor/location
+  do {
+    res = await iterator.next();
+    if (res) {
+      items.push(res);
+    }
+  } while (res !== null);
+  return (
+    items
+      // result is a list of 'up' | 'down', 1:1 per monitor
+      .map(({ groups }) =>
+        // for each location, infer a status
+        groups.reduce<'up' | 'down'>((acc, cur) => {
+          if (acc === 'down') {
+            return acc;
+          }
+          return cur.status === 'down' ? 'down' : 'up';
+        }, 'up')
```

Comment:
if i understand it right, the purpose of this reducer is to get a a status, means you assume it's up, and then if even if there is one down, it will be considered 'down'.

Reply:
`
groups.some(curr => curr.status === 'down') ? 'down' : 'up'
`
will above statement make it more clean? WDYT?

Reply:
Thinking about this, we can can use `find` in preference to `reduce` to search for anything with a down value then immediately return. There's no need to keep iterating once we've found a down value.

Reply:
++ agree with you both here